packages/core/src/components/Toggle/Toggle.styled.tsx

```diff
@@ -24,27 +25,58 @@ export const Wrapper = styled(FieldWithLabel.Field)<ToggleWrapperProps>`
         border-radius: 15px;
     }
 `;
+
 Wrapper.defaultProps = {
     theme: defaultTheme
 };
 
-export const Circle = styled('div')<ToggleWrapperProps>`
-    z-index: 1;
+export const Switch = styled('div')<ToggleWrapperProps>`
+    position: relative;
+    width: 100%;
+    overflow: hidden;
+    display: flex;
+    align-items: flex-end;
+`;
+
+export const OffIcon = styled('div')<ToggleWrapperProps>`
+    color: ${({ theme }) => theme.toggle.iconColor};
+    transition: 0.2s ease-in-out;
     position: absolute;
-    background-color: ${({ theme }) => theme.toggle.bgColor};
+    right: 0;
+    svg {
+        height: ${props => `calc(${getHeight(props)} - 10px) `};
+        path {
+            fill: ${({ theme }) => theme.toggle.iconColor};
+        }
+    }
+`;
 
-    &::after {
-        content: '';
-        margin: 3px;
-        display: block;
-        border-radius: 50%;
-        background: ${({ theme }) => theme.colors.white};
-        box-shadow: 1px 3px 3px 1px rgba(0, 0, 0, 0.2);
-        transition: 0.2s;
-        width: ${props => `calc(${getHeight(props)} - 6px) `};
-        height: ${props => `calc(${getHeight(props)} - 6px) `};
+export const OnIcon = styled('div')<ToggleWrapperProps>`
```

Comment:
You can move few of the common style from OffIcon and OnIcon

Reply:
Done, thanks!